### ISSUE TITLE
Does AbstractService belong in org.threadly.utils

### ISSUE THREAD

**Issue by jentfoo** [threadly/threadly#136]
Right now it is in the org.threadly.concurrent package.  I think I put it there because the thought was these services normally have a thread of their own.  With that said, the way we have implemented "AbstractService", nothing says it _has_ to be a concurrent service.

What do you think @lwahlmeier ....Should we move it to org.threadly.utils to keep the very large concurrent package a bit smaller, or should we just keep it where it is?

If we do move it, I will release a version 3.8.0 which has the current one deprecated and the new one in place, that way the transition to 4.0.0 will be easier.

**Comment by lwahlmeier** [comment]
Hum, I think it probably should be moved, not sure if utils is the right spot either though.  Even if it was not sure its worth doing a 3.8.0 release just for that.  If something else needs to be depriciated maybe but not just for that.

**Comment by jentfoo** [comment]
Yea, right now "utils" is just a dumping ground for small utilities that don't have enough like counterparts to warrant a package on their own.

The only reason I started thinking about it was what we talked about adding a "ServiceInterface".  That starts polluting the already very full org.threadly.concurrent.

I think it's worth a 3.8.0 release, but I will wait to do the release till we see if there is anything else we want to depreciate.  That way we can be sure it's the _last_ release before we go 4.0.0.
